### PR TITLE
Fix Major SonarCloud code smell issues

### PR DESCRIPTION
**Pull request by PeterSuna**
- Add parametrized type to generic
- Merge this if statement with the enclosing one
- Define a constant instead of duplicating this litera
- Remove deprecated class DeclaredEffectiveStatementBase
   Used AbstractDeclaredEffectiveStatement and its subclasses instead
- Use only SLF4J_SIGN_ONLY_FORMAT supress warning with
   justification and only on required methods.
- Use Collections.emptyList() insed of constat.
- Fix retun generic collection from declaredSubstatements().

### CODE REVIEW

File: src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java [hunk 1]
```diff
@@ -469,12 +467,12 @@ private Object translateValueByType(final Object value, final DataSchemaNode nod
 
         checkArgument(node instanceof TypedDataSchemaNode);
         checkArgument(value instanceof String);
-        final TypeAwareCodec xmlCodec = codecs.codecFor((TypedDataSchemaNode) node);
+        final TypeAwareCodec<?,NamespaceContext,?> xmlCodec = codecs.codecFor((TypedDataSchemaNode) node);
```
Comment (vainyksi) [inline src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java:470]: formatting
`TypeAwareCodec<?, NamespaceContext, ?>`

**Review by vainyksi — CHANGES_REQUESTED** [review]
minor changes

File: src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java [hunk 2]
```diff
@@ -50,7 +50,10 @@
 
 public class ModulePrinter {
 
-    private final Set<TypeDefinition> usedTypes;
+    private static final String REFERENCE = "reference";
```
[the remark this answers is not in the document]
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/formats/yang/printer/ModulePrinter.java]: done

File: src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java (hunk 1 above)
Comment (PeterSuna) [reply src/main/java/io/lighty/yang/validator/simplify/stream/TrackingXmlParserStream.java]: done

**Review by vainyksi — APPROVED** [review]

**Review by vainyksi — APPROVED** [review]

**Pull request merged**